restli-common/src/main/java/com/linkedin/restli/common/ComplexResourceKey.java

```diff
@@ -305,11 +322,31 @@ public void makeReadOnly()
                                                                                ComplexKeySpec<?, ?> complexKeyType,
                                                                                ProtocolVersion version)
     throws PathSegmentSyntaxException
+  {
+    return parseString(str, complexKeyType, version, false);
+  }
+
+  /**
+   * Parse the given {@link String} into a {@link ComplexResourceKey}.
+   *
+   * @param str the {@link String} to parse
+   * @param complexKeyType the {@link ComplexKeySpec} of the {@link ComplexResourceKey}
+   * @param version the {@link ProtocolVersion}
+   * @param throwErrorOnValidationFailure if set true throws IllegalArgumentException on validation failure for restli V2
+   * @return a {@link ComplexResourceKey}
+   * @throws PathSegmentSyntaxException
+   */
+  @SuppressWarnings("deprecation")
```

Comment:
Just curious, what's causing this deprecation warning?

Reply:
initially, I had deprecated older methods, removed it

Reply:
removed